src/cmd-cloud-prune

```diff
@@ -141,13 +143,13 @@ def main():
                         # S3().delete_object(args.bucket, f"{args.prefix}{str(current_build.id)}")
                     case "images":
                         raise NotImplementedError
-            if not args.dry_run:
-                build.setdefault("policy-cleanup", []).append("cloud-uploads")
-                builds_json_data["builds"][index] = build
+            build.setdefault("policy-cleanup", []).append("cloud-uploads")
+            builds[len(builds) - 1 - index] = build
```

Comment:
The line `builds_json_data["builds"][index] = build` is exactly where it went wrong. I kept updating the data on wrong index after we switched the enumerating of the builds in **reversed** order which caused us to lose the builds and also switched the ordering.